modules/utxo-lib/src/bitgo/zcash/ZcashPsbt.ts

```diff
@@ -128,9 +129,17 @@ export class ZcashPsbt extends UtxoPsbt<ZcashTransaction<bigint>> {
   // transactions because zcash hashes the value directly. Thus, it is unnecessary to have
   // the previous transaction hash on the unspent.
   signInput(inputIndex: number, keyPair: Signer, sighashTypes?: number[]): this {
+    return this.withUnsafeSignNonSegwitTrue(super.signInput.bind(this, inputIndex, keyPair, sighashTypes));
+  }
+
+  validateSignaturesOfInput(inputIndex: number, validator: ValidateSigFunction, pubkey?: Buffer): boolean {
+    return this.withUnsafeSignNonSegwitTrue(super.validateSignaturesOfInput.bind(this, inputIndex, validator, pubkey));
+  }
+
+  private withUnsafeSignNonSegwitTrue(fn: () => any): any {
```

Comment:
```suggestion
  private withUnsafeSignNonSegwitTrue<T>(fn: () => T): T {
```
Does this work to improve type checking?

Reply:
Yes - this works - and a lot better than how it was before